File: Python/HissingMicrophone.py
# Hissing Microphone https://open.kattis.com/problems/hissingmicrophone

data = list(input())
flag = 0
prev = '0'
for i in range(len(data)):
	if prev == data[i] and prev == 's':
		print("hiss")
		flag = 1
		break
	prev = data[i]
if flag == 0:
	print("no hiss")

File: Java/HissingMicrophone.java
// https://open.kattis.com/problems/hissingmicrophone

import java.util.Scanner;

public class HissingMic {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        char c[] = new char[30];
        c = s.toCharArray();

        for (int i = 0; i < 30; i++) {
            try {
                if (c[i] == c[i + 1] && c[i] == 's') {
                    System.out.println("hiss");
                    break;
                }
            } catch (ArrayIndexOutOfBoundsException e) {
                System.out.println("no hiss");
                break;
            }
        }
    }
}
